File: clients/rust/src/models/rate_limiter_token_bucket_config.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[non_exhaustive]
#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct RateLimiterTokenBucketConfig {
    /// Maximum capacity of the bucket
    pub capacity: u64,

    /// Number of tokens to add per refill interval
    pub refill_amount: u64,

    /// Interval in seconds between refills (minimum 1 second)
    pub refill_interval_seconds: u64,
}

impl RateLimiterTokenBucketConfig {
    pub fn new(capacity: u64, refill_amount: u64, refill_interval_seconds: u64) -> Self {
        Self {
            capacity,
            refill_amount,
            refill_interval_seconds,
        }
    }
}

File: clients/javascript/src/models/rateLimiterTokenBucketConfig.ts
// this file is @generated

export interface RateLimiterTokenBucketConfig {
    /** Maximum capacity of the bucket */
    capacity: number;
    /** Number of tokens to add per refill interval */
    refillAmount: number;
    /** Interval in seconds between refills (minimum 1 second) */
    refillInterval?: number;
}

export const RateLimiterTokenBucketConfigSerializer = {
    _fromJsonObject(object: any): RateLimiterTokenBucketConfig {
        return {
            capacity: object['capacity'],
            refillAmount: object['refill_amount'],
            refillInterval: object['refill_interval'],
        };
    },

    _toJsonObject(self: RateLimiterTokenBucketConfig): any {
        return {
            'capacity': self.capacity,
            'refill_amount': self.refillAmount,
            'refill_interval': self.refillInterval,
        };
    }
}